src/v/cluster/tests/tx_compaction_tests.cc

```diff
@@ -0,0 +1,71 @@
+#include "cluster/rm_stm.h"
+#include "raft/tests/mux_state_machine_fixture.h"
+#include "raft/tests/raft_group_fixture.h"
+#include "storage/tests/utils/disk_log_builder.h"
+#include "tx_compaction_utils.h"
+
+#include <seastar/util/defer.hh>
+
+static ss::logger test_logger{"tx_compaction_tests"};
+
+using cluster::random_tx_generator;
+
+#define STM_BOOTSTRAP()                                                        \
+    storage::ntp_config::default_overrides o;                                  \
+    o.cleanup_policy_bitflags = model::cleanup_policy_bitflags::compaction;    \
+    start_raft(o);                                                             \
+    ss::sharded<cluster::tx_gateway_frontend> tx_gateway_frontend;             \
+    ss::sharded<features::feature_table> feature_table;                        \
+    feature_table.start().get0();                                              \
+    auto stm = ss::make_shared<cluster::rm_stm>(                               \
+      test_logger, _raft.get(), tx_gateway_frontend, feature_table);           \
+    stm->testing_only_disable_auto_abort();                                    \
+    stm->testing_only_enable_transactions();                                   \
+    stm->start().get0();                                                       \
+    auto stop = ss::defer([&] {                                                \
+        _data_dir = "test_dir_" + random_generators::gen_alphanum_string(6);   \
+        stm->stop().get0();                                                    \
+        feature_table.stop().get0();                                           \
+        stop_all();                                                            \
+    });                                                                        \
+    wait_for_confirmed_leader();                                               \
+    wait_for_meta_initialized();                                               \
+    auto log = _storage.local().log_mgr().get(_raft->ntp());                   \
+    log->stm_manager()->add_stm(stm);                                          \
+    BOOST_REQUIRE(log);
+
+storage::disk_log_impl* get_disk_log(storage::log& log) {
+    return dynamic_cast<storage::disk_log_impl*>(log.get_impl());
+}
+
+FIXTURE_TEST(test_tx_compaction_combinations, mux_state_machine_fixture) {
```

Comment:
i really like this test, although i am not sure if i understand why it depends on `mux_state_machine_fixture` ?